daft/datatype.py

```diff
@@ -131,6 +163,24 @@ def struct(cls, fields: dict[str, DataType]) -> DataType:
     def extension(cls, name: str, storage_dtype: DataType, metadata: str | None = None) -> DataType:
         return cls._from_pydatatype(PyDataType.extension(name, storage_dtype._dtype, metadata))
 
+    @classmethod
+    def embedding(cls, name: str, dtype: DataType, size: int) -> DataType:
+        if not isinstance(size, int) or size <= 0:
+            raise ValueError("The size for a embedding must be a positive integer, but got: ", size)
+        return cls._from_pydatatype(PyDataType.embedding(name, dtype._dtype, size))
+
+    @classmethod
+    def image(cls, mode: str | ImageMode, size: tuple[int, int] | None = None) -> DataType:
```

Comment:
Could change `size` to optional `height` and `width` args, but didn't think that the UX of an extra arg was worth it, and the `size = (height, width)` tuple is pretty common in image APIs AFAIK. @samster25 Lmk if you disagree, I'd defer to you on this one!

Reply:
I would say we should make it explicit with height and width. some libraries follow matrix notation (height, width) and other libraries following (x,y) notation.
https://docs.opencv.org/3.4/da/d6e/tutorial_py_geometric_transformations.html
https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.resize